packages/react-output-target/react-component-lib/createComponent.tsx

```diff
@@ -80,7 +80,7 @@ export const createReactComponent = <
         style,
       };
 
-      return React.createElement(tagName, newProps, children);
+      return createElement(tagName, newProps, children);
```

Comment:
This feels like it could be easily regressed by someone that doesn't have background knowledge of why we made this change. It may be worthwhile to add a small explainer comment stating we need to use `createElement` in this way + a link to the issue you opened up in the Vite repo